### ISSUE TITLE
System.PlatformNotSupportedException: System.Data.SqlClient is not supported on this platform

### ISSUE THREAD

**Issue by thomasvdb** [Azure/azure-functions-host#5832]
#### Investigative information
- Timestamp: 2020-03-26 09:48:09.304
- Function App version: 3.0.13139.0
- Invocation ID: 5da3c701a5a9704fac16e60bc400e607
- Region: West-Europe

After deploying our Azure Function (which was recently upgrade from .NET Core 2.2 to .NET Core 3.1), we sometimes receive the following error:
```
An exception occurred in the database while iterating the results of a query for context type 
'...SourceContext'.
System.PlatformNotSupportedException: System.Data.SqlClient is not supported on this platform
```

If we then switch the Runtime version of the Azure Function to v2 and back to v3; the issue is solved.
What strikes me when watching the logs is the following entry when the error occurs:
```
Entity Framework Core 2.1.1-rtm-30846 initialized '...SourceContext' using provider 'Microsoft.EntityFrameworkCore.SqlServer' with options: None
```

I've checked and the DLLs in the ZIP are all 3.1 DLLs for Entity Framework Core.
We also do a clean during the build to prevent any leftover DLLs.

#### Repro steps
This is our YAML file for the App Service Deploy of the Function (values are changed to dummy values).

```
variables:
  functionsAppName: 'ourapplicationname'

steps:
- task: AzureRmWebAppDeployment@4
  displayName: 'Azure App Service Deploy: $(functionsAppName)'
  inputs:
    azureSubscription: 'subscriptionname'
    appType: functionApp
    WebAppName: '$(functionsAppName)'
    packageForLinux: '$(System.DefaultWorkingDirectory)/ProjectName/drop/ProjectName.zip'
```

#### Actual behavior
Exception thrown by querying our database in the Azure Function.
```
An exception occurred in the database while iterating the results of a query for context type 
'...SourceContext'.
System.PlatformNotSupportedException: System.Data.SqlClient is not supported on this platform
```

#### Known workarounds
Switch to Azure Function v2 in the portal and back to v3 right away.

#### Related information 
- .NET Core 3.1
- Azure Function is written in .NET and uses a Service Bus trigger
- Entity Framework Core 3.1.1
- Microsoft.NET.Sdk.Functions 3.0.5

**Comment by yojagad** [comment]
Can you try upgrading to Microsoft.Data.SqlClient and check to see if you still continue to see this error? I'm basing this off of similar, longer threads where some people had this issue with System.Data.SqlClient. https://github.com/Azure/Azure-Functions/issues/1370 and https://github.com/dotnet/SqlClient/issues/289

**Comment by thomasvdb** [comment]
I've read these threads before but when looking in the NuGet package manager for our solution this library wasn't listed as NuGet package.

When doing a text search in our solution I did found one reference though.
Adding the NuGet package as we speak and I'll monitor it the coming days (as the issue doesn't pop up on every deploy).

**Comment by pavinan** [comment]
We are also facing same issue. But downgrading `System.Data.SqlClient` to 4.5.0 works.

**Comment by rickfunaki** [comment]
This was a significant issue for us and took down crucial portions of our production environment. We had to fall back to business continuity procedures to work around the outage in all of our functions until we could find a code workaround.

This had happened at the end of January as well (previous GitHub issue: https://github.com/Azure/azure-functions-host/issues/5414), we were able to work around then by pinning to a previous Functions runtime version, but this time we were not able to pin to any previous version that did not exhibit the problem. We had been able to use 2.0.12888 but after the latest release to 2.0.13017.0 we could not pin back that far and every version we could pin to exhibited this problem.

I just cannot believe that these types of breaking changes would be allowed to go out automatically with no warning, and that there is no recourse for correction to the customer; and not only that, that this happen twice in a few months! Azure is really dropping the ball here, the last time this happened it took a month for a fix to be pushed, but that later release fix never did work for us.

Things we tried that did not work:
- Pinning to any previous version of the Azure Functions runtime via application setting override
- Re-provisioning the functions resources in hopes that we get one that hadn't been upgraded yet, allowing us to pin farther back
- Upgrading all of our System.Data.SqlClient code references and NuGet package references to Microsoft.Data.SqlClient (still same issue as reported above, after switching it was then just stating _that_ reference was supported)
- Upgrading the Azure Functions SDK to the latest version

What ended up working for us was the below:
- Revert Functions SDK to 1.0.29 (from 1.0.31)
- Keep System.Data.SqlClient but downgrade to 4.5.0 (from 4.5.1)
